solr/core/src/test/org/apache/solr/update/processor/AtomicUpdatesJsonTest.java

```diff
@@ -0,0 +1,206 @@
+
+    doc = new SolrInputDocument();
+    doc.setField("id", "1");
+    doc.setField(
+        "name",
+        new HashMap<String, Object>() {
+          {
+            put("add", new String[] {"ddd", "eee"});
+            put("remove", new String[] {"aaa", "ccc"});
+          }
+        });
+    updateJ(jsonAdd(doc), null);
+    assertU(commit());
+    assertQ(req("q", "name:aaa", "indent", "true"), "//result[@numFound = '0']");
+    assertQ(req("q", "name:ccc", "indent", "true"), "//result[@numFound = '0']");
+    assertQ(req("q", "name:bbb", "indent", "true"), "//result[@numFound = '1']");
+    assertQ(req("q", "name:ddd", "indent", "true"), "//result[@numFound = '1']");
+    assertQ(req("q", "name:eee", "indent", "true"), "//result[@numFound = '1']");
+  }
+
+  @Test
+  public void testAddAndRemove() throws Exception {
+    SolrInputDocument doc = new SolrInputDocument();
+    doc.setField("id", "1");
+    doc.setField("name", new String[] {"aaa", "bbb", "ccc"});
+    updateJ(jsonAdd(doc), null);
+    assertU(commit());
+    assertQ(req("q", "name:aaa", "indent", "true"), "//result[@numFound = '1']");
+    assertQ(req("q", "name:bbb", "indent", "true"), "//result[@numFound = '1']");
+    assertQ(req("q", "name:ccc", "indent", "true"), "//result[@numFound = '1']");
+
+    doc = new SolrInputDocument();
+    doc.setField("id", "1");
+    doc.setField("name", Map.of("add", "ddd", "remove", "bbb"));
+    updateJ(jsonAdd(doc), null);
+    assertU(commit());
+    assertQ(req("q", "name:ddd", "indent", "true"), "//result[@numFound = '1']");
+    assertQ(req("q", "name:bbb", "indent", "true"), "//result[@numFound = '0']");
+    assertQ(req("q", "name:ccc", "indent", "true"), "//result[@numFound = '1']");
+    assertQ(req("q", "name:aaa", "indent", "true"), "//result[@numFound = '1']");
+  }
+
+  @Test
+  public void testAtomicUpdateModifierNameSingleValued() throws Exception {
+    // Testing atomic update with a single-valued field named 'set'
+    SolrInputDocument doc = new SolrInputDocument();
+    doc.setField("id", "1");
+    doc.setField("set", "setval");
+    doc.setField("name", new String[] {"aaa"});
+    updateJ(jsonAdd(doc), null);
+    assertU(commit());
+    assertQ(req("q", "set:setval", "indent", "true"), "//result[@numFound = '1']");
+
+    doc = new SolrInputDocument();
+    doc.setField("id", "1");
+    doc.setField("set", Map.of("set", "modval"));
+    updateJ(jsonAdd(doc), null);
+    assertU(commit());
+    assertQ(req("q", "set:modval", "indent", "true"), "//result[@numFound = '1']");
+    assertQ(req("q", "set:setval", "indent", "true"), "//result[@numFound = '0']");
+
+    doc = new SolrInputDocument();
+    doc.setField("id", "1");
+    doc.setField(
+        "set",
+        new HashMap<String, String>() {
+          {
+            put("set", null);
+          }
+        });
+    updateJ(jsonAdd(doc), null);
+    assertU(commit());
+    assertQ(req("q", "set:modval", "indent", "true"), "//result[@numFound = '0']");
```

Comment:
Don't need indent:true in any of this file.  I recall Solr does the indenting anyway in a test failure

Reply:
Updated all.